gsa/src/web/pages/schedules/component.js

```diff
@@ -15,151 +15,161 @@
+        type: 'setState',
+        newState: {
+          comment: undefined,
+          dialogVisible: true,
+          duration: undefined,
+          freq: undefined,
+          id: undefined,
+          interval: undefined,
+          monthdays: undefined,
+          name: undefined,
+          startDate: undefined,
+          timezone,
+          title: undefined,
+          weekdays: undefined,
+        },
       });
     }
 
-    this.handleInteraction();
-  }
+    handleInteraction();
+  };
 
-  closeScheduleDialog() {
-    this.setState({dialogVisible: false});
-  }
+  const closeScheduleDialog = () => {
+    dispatch({type: 'setState', newState: {dialogVisible: false}});
+  };
 
-  handleCloseScheduleDialog() {
-    this.closeScheduleDialog();
-    this.handleInteraction();
-  }
+  const handleCloseScheduleDialog = () => {
+    closeScheduleDialog();
+    handleInteraction();
+  };
 
-  handleInteraction() {
-    const {onInteraction} = this.props;
+  const handleInteraction = () => {
     if (isDefined(onInteraction)) {
       onInteraction();
     }
-  }
-
-  render() {
-    const {
-      children,
-      onCloned,
-      onCloneError,
-      onCreated,
-      onCreateError,
-      onDeleted,
-      onDeleteError,
-      onDownloaded,
-      onDownloadError,
-      onInteraction,
-      onSaved,
-      onSaveError,
-    } = this.props;
-
-    const {dialogVisible, ...dialogProps} = this.state;
-
-    return (
-      <EntityComponent
-        name="schedule"
-        onCreated={onCreated}
-        onCreateError={onCreateError}
-        onCloned={onCloned}
-        onCloneError={onCloneError}
-        onDeleted={onDeleted}
-        onDeleteError={onDeleteError}
-        onDownloaded={onDownloaded}
-        onDownloadError={onDownloadError}
-        onInteraction={onInteraction}
-        onSaved={onSaved}
-        onSaveError={onSaveError}
-      >
-        {({save, ...other}) => (
-          <React.Fragment>
-            {children({
-              ...other,
-              create: this.openScheduleDialog,
-              edit: this.openScheduleDialog,
-            })}
-            {dialogVisible && (
-              <ScheduleDialog
-                {...dialogProps}
-                onClose={this.handleCloseScheduleDialog}
-                onSave={d => {
-                  this.handleInteraction();
-                  return save(d).then(() => this.closeScheduleDialog());
-                }}
-              />
-            )}
-          </React.Fragment>
-        )}
-      </EntityComponent>
-    );
-  }
-}
+  };
+
+  const handleSaveSchedule = data => {
+    handleInteraction();
+
+    const {id, ...other} = data;
+
+    if (isDefined(id)) {
+      return modifySchedule(data)
+        .then(onSaved, onSaveError)
+        .then(() => closeScheduleDialog());
+    }
+
+    return createSchedule(other)
+      .then(onCreated, onCreateError)
+      .then(() => closeScheduleDialog());
+  };
+
+  const {dialogVisible, ...dialogProps} = state;
```

Comment:
The `dialogProps` is just our name for any props that isn't `dialogVisible`. They're mostly set when you open schedule dialog.